File: data/readable_format/fortran/file962.f90
! combined.f90
module mod_setprecision
  implicit none
  integer, parameter :: sprec = kind(1.0)
  integer, parameter :: dprec = kind(1.0d0)
end module mod_setprecision

program test_mod_setprecision
  use mod_setprecision
  implicit none

  ! Test for single precision
  if (sprec .eq. kind(1.0)) then
    print *, "Test for single precision (sprec) passed."
  else
    print *, "Test for single precision (sprec) failed."
  endif

  ! Test for double precision
  if (dprec .eq. kind(1.0d0)) then
    print *, "Test for double precision (dprec) passed."
  else
    print *, "Test for double precision (dprec) failed."
  endif
end program test_mod_setprecision

File: data/readable_format/cpp/file962.cpp
// combined.cpp
#include <iostream>
#include <cmath>

namespace mod_setprecision {
    constexpr int sprec = sizeof(float);
    constexpr int dprec = sizeof(double);
}

int main() {
    bool tests_passed = true;

    if (mod_setprecision::sprec != sizeof(float)) {
        std::cout << "Test for single precision failed." << std::endl;
        tests_passed = false;
    }

    if (mod_setprecision::dprec != sizeof(double)) {
        std::cout << "Test for double precision failed." << std::endl;
        tests_passed = false;
    }

    if (tests_passed) {
        std::cout << "All tests passed." << std::endl;
    } else {
        std::cout << "Some tests failed." << std::endl;
    }

    return !tests_passed;
}